### PR TITLE
get rely running and clean up build scripts a little bit

### PR DESCRIPTION
**Pull request by bandersongit**
Basically everything except snapshot matchers is working in the browser and with node.

See the tests directory for a couple example tests.

image

I add a runTests script and aliased the "test" script to that. I also changed the build script to use dune instead of JSOO explicitly so you don't need to create incremental targets/dune handles linking for you.

### CODE REVIEW

File: jsStubs/unix_isatty.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+//Provides: unix_isatty
+
+function unix_isatty (f) {
+    return false;
```
Comment (aweis) [inline jsStubs/unix_isatty.js:4]: nice

File: runTests.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+esy buildTestsJs &&
```
Comment (aweis) [inline runTests.sh:3]: nice

File: tests/TestFramework.re [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  let readSnapshotNames = _ => [];
+  let readFile: string => string = s => s;
+  let removeFile: string => unit = _ => ();
+  let writeFile: (string, string) => unit = (_, _) => ();
+  let existsFile: string => bool = _ => false;
+  let mkdirp = _ => ();
+};
+/* Snapshots will be virtual-modulified at some point, until then we need to inject
+ * something custom with RelyInternal to make JSOO happy because the default snapshot
+ * implementation uses the Unix module  */
```
Comment (aweis) [inline tests/TestFramework.re:11]: I may be missing something about rely - do we need to have snapshotting work with JSOO?  Testing would not happen in the browser

**Review by aweis — CHANGES_REQUESTED** [review]
Can you please pull in the upstream changes so and resubmit so I can merge this?  Very cool work!

File: tests/TestFramework.re (hunk 3 above)
Comment (bandersongit) [reply tests/TestFramework.re:11]: Snapshots aren't dependent on the browser, basically it just checks some string output versus a file and provides a hook to create/update the file if it doesn't exist/the test fails. You may not need them at all for testing irmin, but for things like code gen/pretty printers/anything that produces really long string output it's a nice feature to have that I want to make sure eventually works

**Pull request merged**